### PR TITLE
Pic 850 message signing

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        cryptoFactoryBean.setKeyStoreLocation(FileSystemResource(keystoreFilePath))
+        return cryptoFactoryBean
+    }
+
+    @Bean
+    @Throws(Exception::class)
+    fun securityInterceptor(): Wss4jSecurityInterceptor {
+        val securityInterceptor = Wss4jSecurityInterceptor()
+
+        val cryptoBean = getCryptoFactoryBean().getObject()
```
Comment (jhevans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt:45]: Is this Bean used elsewhere? Just wondering if we need the @Bean annotation at line 31?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        // validate incoming request
+        securityInterceptor.setValidationActions(requestActions)
+        securityInterceptor.setValidationSignatureCrypto(cryptoBean)
+        securityInterceptor.setValidationDecryptionCrypto(cryptoBean)
+        securityInterceptor.setValidationCallbackHandler(securityCallbackHandler())
```
Comment (jhevans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt:51]: Same question re @Bean

**Review by jhevans — APPROVED** [review]
A couple of tiny queries to cast your eyes over but all good 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt (hunk 1 above)
Comment (chris-faulkner-mojdt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt:45]: Yes - securityInterceptor is used by InterceptorConfig. InterceptorConfig implements WsConfigurerAdapter and puts that bean in place to decrypt incoming requests and sign / encrypt the response.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt (hunk 2 above)
Comment (chris-faulkner-mojdt) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/application/WebServiceSecurityConfig.kt:51]: Yes - the securityInterceptor is needed and needs that callback

**Pull request merged**